### PR TITLE
Closes #285

### LINKED ISSUE #285: Create `h_dose_grid_range` function or slot.
Create `h_dose_grid_range` function or slot in `Data` class and use it in `nextBest` methods (particular `nextBestTD` and `nextBestTDsamples`).

### PR DESCRIPTION
**Pull request by wwojciech**
Closes #285

### CODE REVIEW

File: R/Rules-methods.R [hunk 1]
```diff
@@ -865,14 +865,10 @@ setMethod(
       next_dose = next_dose_drt
     )
 
-    dose_grid_range <- c(
-      data@doseGrid[ifelse(data@placebo && data@nGrid >= 2, 2, 1)],
-      data@doseGrid[data@nGrid]
-    )
-    if (!h_in_range(dose_target_drt, range = dose_grid_range, bounds_closed = TRUE) && !SIM) {
+    if (!h_in_range(dose_target_drt, range = h_dose_grid_range(data), bounds_closed = TRUE) && !SIM) {
```
Comment (danielinteractive) [inline R/Rules-methods.R:868]: unrelated but: the `SIM` variable does not follow conventions, can we use something else? e.g. `do_sim` or so?

File: R/helpers.R [hunk 2]
```diff
@@ -653,6 +653,33 @@ h_plot_data_cohort_lines <- function(cohort,
...
+#' Getting the Dose Grid Range
+#'
+#' @description `r lifecycle::badge("stable")`
+#'
+#' This helper function gets the dose range for a given data object.
+#' It returns `c(-Inf, Inf)` if the range cannot be determined. This happens
+#' if and only if dose grid is empty or there is only placebo in the grid.
+#'
+#' @param data (`Data`)\cr a data object that contains a dose grid.
+#'
```
Comment (danielinteractive) [inline R/helpers.R:665]:
```suggestion
#' @return [details]
```

**Review by danielinteractive — COMMENTED** [review]
thx Wojciech! please see comments inside

File: R/Rules-methods.R (hunk 1 above)
Comment (wwojciech) [reply R/Rules-methods.R:868]: It seems that this would need to be changed in a few places - so I will create a new issue for this.

**Review by danielinteractive — APPROVED** [review]

**Pull request merged**